lib/iris/fileformats/netcdf.py

```diff
@@ -821,9 +828,15 @@ def _create_cf_dimensions(self, cube, dimension_names,
             None.
 
         """
+        msg = ('NetCDF saving currently assigns the outermost dimensions to '
+               'unlimited. This behaviour is to be deprecated, in favour of '
+               'no automatic assignment. To switch to the new behaviour, set '
+               'iris.FUTURE.netcdf_no_unlimited to True.')
+
         unlimited_dim_names = []
         if unlimited_dimensions is None:
-            if dimension_names:
+            if dimension_names and not iris.FUTURE.netcdf_no_unlimited:
+                warnings.warn(msg)
```

Comment:
As [mentioned previously](https://github.com/SciTools/iris/pull/1452#issuecomment-64332062), the deprecation warning should only come from the public interfaces.